pyblish/util.py

```diff
@@ -26,13 +26,59 @@ def publish(context=None, plugins=None, targets=None):
             defaults to results of discover()
         targets (list, optional): Targets to include for publish session.
 
+    Returns:
+        Context: The context processed by the plugins.
+
     Usage:
         >> context = plugin.Context()
         >> publish(context)     # Pass..
         >> context = publish()  # ..or receive a new
 
     """
 
+    context = api.Context() if context is None else context
+
+    # Since the Context object is a singleton, we can just run though the
```

Comment:
Oh, it's not a singleton. Every context is unique. There's no need for this comment either IMO.

Reply:
Done :)